### DISCUSSION TITLE
MUI Select with react-hook-form setValue not working

### DISCUSSION THREAD

**Discussion by mib-kamel** [react-hook-form/react-hook-form#8544] [Q&A] [answered]
Hello,

I have a project using MUI Select and react-hook-form 7 with YUP resolver

When I try to change the value of a field registered in MUI select by using setValue(....) there is no change happening

[Code Example](https://codesandbox.io/s/cranky-river-hmdpec?file=/src/App.js)

Any help please?

**Comment by bluebill1049** [comment, marked as answer]
MUI select is a controlled component as far as i know, so it's better to wrapped with Controller or hooked with useController.

**Reply by mib-kamel** [reply]
Worked, Thank you.

[Solution](https://codesandbox.io/s/admiring-curie-stss8q?file=/src/App.js)

**Reply by bluebill1049** [reply]
nice 👍

**Reply by garrettg123** [reply]
The solution no longer works. See the codesandbox link above.
